src/coreclr/vm/JitQCallHelpers.h

```diff
@@ -22,4 +22,71 @@ extern "C" void * QCALLTYPE ResolveVirtualFunctionPointer(QCall::ObjectHandleOnS
 extern "C" CORINFO_GENERIC_HANDLE QCALLTYPE GenericHandleWorker(MethodDesc * pMD, MethodTable * pMT, LPVOID signature, DWORD dictionaryIndexAndSlot, Module* pModule);
 extern "C" void QCALLTYPE InitClassHelper(MethodTable* pMT);
 
+#ifdef TARGET_32BIT
+extern "C" int32_t QCALLTYPE Math_ActualDivisionInt(int32_t i, int32_t j)
```

Comment:
I'm unclear why these aren't just FCalls. They satisfy all the requirements and will be lower overhead and as fast as can be - minus doing all the work in the JIT.

Reply:
Something like this (without the helper frame): https://github.com/dotnet/runtime/blob/8b0448f9eedf6c63ec381ce32b6e640dcb87bb3c/src/coreclr/vm/jithelpers.cpp#L2395

Reply:
This depends on SoftwareExceptionFrame that is not much different from helper method frame. It requires unwinding through C++ code. We have discussed this problem in #108480. I expect that the eventual solution will be to implement this in asm code, same as what we do for naot (it will avoid the need for unwinding through C++ code).